microsoft/PowerToys
# [PowerToys Run] Add everything plugin, port it to .NET Core 3.0

## Summary of the Pull Request

As referenced in #3278, #4318, #1791, and some duplicates, there is a want to restore the Everything integration to PowerToys Run

My main reasoning for wanting Everything is a little different, the Windows search API could be faster but is 'OK'.  As these tools are meant for power users, my issue with the windows search in PT Run is that there's some places it just won't go.  (e.g. System32, other system folders...)

Having Everything to scan the MFT as a fallback is a must, if you truely want to find 'everything' :).

 
## References

#3278, #4318, #1791, #3303


## PR Checklist
* [x] Applies to #3278
* [x] Tests added/passed - ran as many as possible
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #3278


## Detailed Description of the Pull Request / Additional comments

- Ported the Everything plugin to .NET Core 3.0 and integrated it with PowerToys Run
- Made the Context menu consistent with the Folder plugin


## Validation Steps Performed

Ran the dev and release versions locally. Seems to function well enough. Not surprisingly, as it's ported straight from jjw24/Wox and that has been running on my system for ages (and this implementation is still basically Wox).

No conflicts with Microsoft.Plugin.Folder that I could see (oddly enough, I expected duplicated results and such...?). 
The fastest result seems to win. Probably someone in the team who knows it off the top of their head could clarify why the search results don't clash, because I couldn't be bothered to check. Why look a gift horse in the mouth, right? 

In my fooling around, it seems that Everything nicely fills in the gaps of non-indexed (or not yet indexed) files and locations.

## To-Do

* [ ] Could probably use a cleanup. The images folder is entirely useless now, as we use glyphs
* [ ] I'll see if I can't get translations working.
